### ISSUE TITLE
[V0.31.0] `no-bare-urls` reports false positive

### ISSUE THREAD

**Issue by regseb** [DavidAnson/markdownlint#961]
The new version 0.31.0 of Markdownlint reports a false positive for the following code: [demo](https://dlaa.me/markdownlint/#%25m%23%20Test%20case%0A%0A%3Ca%20href%3D%22https%3A%2F%2Ffoo.com%22%0A%3EFalse-positive%3C%2Fa%3E%0A%0A%3C!--%20markdownlint-disable%20no-inline-html%20--%3E%0A%0A%3Ca%20href%3D%22https%3A%2F%2Ffoo.com%22%3EGood%3C%2Fa%3E%0A%3Ca%0Ahref%3D%22https%3A%2F%2Ffoo.com%22%3EGood%3C%2Fa%3E%0A%3Ca%20href%3D%22https%3A%2F%2Ffoo.com%22%3E%0AGood%3C%2Fa%3E%0A) / [reality](https://github.com/regseb/metalint/blob/38e7c23/README.md?plain=1#L103-L105)

```Markdown
<a href="https://foo.com"
>False-positive</a>
```

> [MD034 / no-bare-urls](https://github.com/DavidAnson/markdownlint/blob/v0.31.0/doc/md034.md) Bare URL used [Context: "http://foo/com"]

I think it's the HTML detection that's broken in version 0.31.0, as the `no-inline-html` rule isn't reported for the above code.

**Comment by DavidAnson** [comment]
As your demo link shows, that syntax is not recognized as HTML and renders as if it were Markdown text. Therefore, I claim it is correct that the "inline HTML" rule does not warn about it and also correct that the embedded URL is flagged as by the "bare URL" rule.
